src/utils/geometry.rs

```diff
@@ -1122,6 +1122,30 @@ impl<N: Coordinate, Kind> Rectangle<N, Kind> {
         }
     }
 
+    /// Create a new [`Rectangle`] from the coordinates of its top-left corner and its dimensions
+    #[inline]
+    pub fn new(loc: Point<N, Kind>, size: Size<N, Kind>) -> Self {
+        Rectangle { loc, size }
+    }
+
+    /// Create a new [`Rectangle`] from its dimensions, with location zero
+    #[inline]
+    pub fn from_size(size: Size<N, Kind>) -> Self {
+        Rectangle {
+            loc: (N::ZERO, N::ZERO).into(),
+            size,
+        }
+    }
+
+    /// Create a new [`Rectangle`] with location and size zero
+    #[inline]
+    pub fn zero() -> Self {
```

Comment:
I feel like this function should be marked `const`. Any reason against that?

Reply:
Good point. The `euclid` version isn't, I guess since https://docs.rs/num-traits/latest/num_traits/identities/trait.Zero.html#tymethod.zero isn't (which perhaps should be, though that may let it be used with heap-allocated bigints as scalars, or such).

If we want to move to `euclid`, it makes sense not to add `const` where `euclid` doesn't have it. If not, we may as well.

Though we also don't seem to have a single `const fn` in `geometry.rs` currently. And I guess no one has complained about that so far.

Reply:
Fine to skip this then.